src/main/target/SITL/target.c

```diff
@@ -324,11 +404,17 @@ void pwmShutdownPulsesForAllMotors(uint8_t motorCount) {
 void pwmCompleteMotorUpdate(uint8_t motorCount) {
 	UNUSED(motorCount);
 	// send to simulator
-	// for gazebo8 ArduCopterPlugin remap, range = [0.0, 1.0]
-	pwmPkt.motor_speed[3] = motorsPwm[0] / 1000.0f;
-	pwmPkt.motor_speed[0] = motorsPwm[1] / 1000.0f;
-	pwmPkt.motor_speed[1] = motorsPwm[2] / 1000.0f;
-	pwmPkt.motor_speed[2] = motorsPwm[3] / 1000.0f;
+	// for gazebo8 ArduCopterPlugin remap, normal range = [0.0, 1.0], 3D rang = [-1.0, 1.0]
+
+	double outScale = 1000.0;
+	if (feature(FEATURE_3D)) {
+		outScale = 500.0;
+	}
+
+	pwmPkt.motor_speed[3] = motorsPwm[0] / outScale;
```

Comment:
Different motor numbering between gazebo and BF will be confusing. Isn't there some way to use identical motor numbers (maybe special simulator mixer?)